btcec/schnorr/musig2/musig2_test.go

```diff
@@ -439,3 +439,83 @@ func (mr *memsetRandReader) Read(buf []byte) (n int, err error) {
 	}
 	return len(buf), nil
 }
+
+// TestSigningWithAggregatedNonce tests the aggregated nonce signing flow where
+// a coordinator aggregates nonces and distributes the combined nonce to
+// participants, rather than each participant aggregating nonces themselves.
+func TestSigningWithAggregatedNonce(t *testing.T) {
```

Comment:
We should also add tests for the input validation edge cases:
  * Call to `RegisterCombinedNonce` fails if we have a combined nonce, or already partially registered some of the 3rd party nonces.